From fb2723595a7bee3a69596b60ce8038d5cee85384 Mon Sep 17 00:00:00 2001
From: Zhao Yihan <zhaoyihan@wxiat.com>
Date: Wed, 11 May 2022 15:13:59 +0800
Subject: [PATCH] sw64: map address by OR operation in __va()
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: fb2723595a7bee3a69596b60ce8038d5cee85384
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56OLG

--------------------------------

Signed-off-by: Zhao Yihan <zhaoyihan@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/include/asm/page.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/sw_64/include/asm/page.h b/arch/sw_64/include/asm/page.h
index 6e17d5e437c5..363785c9f082 100644
--- a/arch/sw_64/include/asm/page.h
+++ b/arch/sw_64/include/asm/page.h
@@ -46,7 +46,7 @@ extern unsigned long __phys_addr(unsigned long);
 #endif
 
 #define __pa(x)			__phys_addr((unsigned long)(x))
-#define __va(x)			((void *)((unsigned long) (x) + PAGE_OFFSET))
+#define __va(x)			((void *)((unsigned long) (x) | PAGE_OFFSET))
 #define virt_to_page(kaddr)	pfn_to_page(__pa(kaddr) >> PAGE_SHIFT)
 #define virt_addr_valid(kaddr)	pfn_valid(__pa(kaddr) >> PAGE_SHIFT)
 
-- 
2.33.0

